packages/interface/src/components/complementary-area-header/style.scss

```diff
@@ -35,12 +36,11 @@
 			display: flex;
 		}
 	}
-}
 
-// This overrides the negative margins between two consecutives panels.
-// since the first panel is hidden.
-.components-panel__header + .interface-complementary-area-header {
-	@include break-medium() {
-		margin-top: 0;
+	// Emulate 'compact' button size, as Dropdown does not support compact yet.
+	.components-dropdown > .components-button.has-icon {
+		height: $grid-unit-40;
+		min-width: $grid-unit-40;
+		padding: $grid-unit-05;
```

Comment:
This doesn’t seem like a bad idea because it should ensure any icon-having-dropdown-button in the complementary area headers will be consistent regardless of whether someone implements one whose button wasn’t `size="compact"`. However, I do want to point out that `DropdownMenu` does support `toggleProps` and that can be used to pass along the `size` prop.
<details><summary>Diff to use `toggleProps` and not style override</summary>

```diff
diff --git a/packages/edit-site/src/components/global-styles/ui.js b/packages/edit-site/src/components/global-styles/ui.js
index b50e09550f..09c1a8b341 100644
--- a/packages/edit-site/src/components/global-styles/ui.js
+++ b/packages/edit-site/src/components/global-styles/ui.js
@@ -76,7 +76,11 @@ function GlobalStylesActionMenu() {
 
 	return (
 		<GlobalStylesMenuFill>
-			<DropdownMenu icon={ moreVertical } label={ __( 'More' ) }>
+			<DropdownMenu
+				icon={ moreVertical }
+				label={ __( 'More' ) }
+				toggleProps={ { size: 'compact' } }
+			>
 				{ ( { onClose } ) => (
 					<>
 						<MenuGroup>
diff --git a/packages/interface/src/components/complementary-area-header/style.scss b/packages/interface/src/components/complementary-area-header/style.scss
index 8cd2a5523c..1e5a000a1b 100644
--- a/packages/interface/src/components/complementary-area-header/style.scss
+++ b/packages/interface/src/components/complementary-area-header/style.scss
@@ -36,11 +36,4 @@
 			display: flex;
 		}
 	}
-
-	// Emulate 'compact' button size, as Dropdown does not support compact yet.
-	.components-dropdown > .components-button.has-icon {
-		height: $grid-unit-40;
-		min-width: $grid-unit-40;
-		padding: $grid-unit-05;
-	}
 }

```
</details>

If we don’t want to use `toggleProps`, I'd at least suggest the comment here be updated to express that the point is to enforce sizing consistency. Saying that Dropdown doesn’t yet support compact isn’t accurate (for `Dropdown` it’s n/a as the consumer supplies the toggle button, for `DropdownMenu` it’s supported through `toggleProps`).

Reply:
I would also recommend using toggleProps. I think it is better to avoid hard-coding the size and use the API available in the component as much as possible.

Reply:
Updated with `toggleProps` in https://github.com/WordPress/gutenberg/pull/61331/commits/930e082fe299b604adb8ac6d6c80986030e9858f. Thanks!